subsys/bluetooth/host/gatt.c

```diff
@@ -2068,6 +2068,20 @@ static struct net_buf *nfy_mult[CONFIG_BT_MAX_CONN];
 static int gatt_notify_mult_send(struct bt_conn *conn, struct net_buf **buf)
 {
 	int ret;
+	uint16_t first_attr_len = sys_get_le16(&((*buf)->data[3]));
+
+	/* Convert to ATT_HANDLE_VALUE_NTF if containing a single handle. */
+	if ((*buf)->len == (1 + sizeof(struct bt_att_notify_mult) + first_attr_len)) {
+		uint16_t handle = sys_get_le16(&((*buf)->data[1]));
+		/* Remove the ATT_MULTIPLE_HANDLE_VALUE_NTF opcode,
+		 * attribute handle and length
+		 */
+		(void)net_buf_pull(*buf, 1 + sizeof(struct bt_att_notify_mult));
+		/* Add back an ATT_HANDLE_VALUE_NTF opcode and attr handle */
+		net_buf_push_le16(*buf, handle);
+		net_buf_push_u8(*buf, BT_ATT_OP_NOTIFY);
+		BT_DBG("Converted BT_ATT_OP_NOTIFY_MULT with single attr to BT_ATT_OP_NOTIFY");
+	}
```

Comment:
Took me a while to grasp what was going on here. 

I think this part can be improved in terms of readability by using some more intermediate variables, so that we can avoid expressions like `&((*buf)->data[3])` which aren't very pleasing to the eye. 

I think it can be done in a way that doesn't make it any less efficient.

Reply:
yep definitely.